core/src/main/java/org/apache/gravitino/cache/ReverseIndexRules.java

```
@@ -0,0 +1,147 @@
+                        Namespace nsTable = Namespace.fromString(securableObject.parent());
+                        namespace =
+                            NamespaceUtil.ofTable(
+                                roleEntity.namespace().level(0),
+                                nsTable.level(0),
+                                nsTable.level(1));
+                        break;
+                      case TOPIC:
+                        entityType = Entity.EntityType.TOPIC;
+                        Namespace nsTopic = Namespace.fromString(securableObject.parent());
+                        namespace =
+                            NamespaceUtil.ofTopic(
+                                roleEntity.namespace().level(0),
+                                nsTopic.level(0),
+                                nsTopic.level(1));
+                        break;
+                      case MODEL:
+                        entityType = Entity.EntityType.MODEL;
+                        Namespace nsModel = Namespace.fromString(securableObject.parent());
+                        namespace =
+                            NamespaceUtil.ofModel(
+                                roleEntity.namespace().level(0),
+                                nsModel.level(0),
+                                nsModel.level(1));
+                        break;
+                      case FILESET:
+                        entityType = Entity.EntityType.FILESET;
+                        Namespace nsFileset = Namespace.fromString(securableObject.parent());
+                        namespace =
+                            NamespaceUtil.ofFileset(
+                                roleEntity.namespace().level(0),
+                                nsFileset.level(0),
+                                nsFileset.level(1));
+                        break;
+                      default:
+                        LOG.info("Unprocessed securable object type: " + securableObject.type());
```

I'm afraid that we may need to throw an exception here, or it will go to L40 and continue to run the rest of the code.